app/src/main/java/org/jellyfin/mobile/fragment/WebViewFragment.kt

```diff
@@ -190,16 +190,14 @@ class WebViewFragment : Fragment() {
     }
 
     fun onSelectServer(error: Boolean = false) {
-        activity?.run {
-            if (lifecycle.currentState.isAtLeast(Lifecycle.State.RESUMED)) {
-                if (error) {
-                    val extras = Bundle().apply {
-                        putBoolean(Constants.FRAGMENT_CONNECT_EXTRA_ERROR, true)
-                    }
-                    replaceFragment<ConnectFragment>(extras)
-                } else {
-                    addFragment<ConnectFragment>()
+        if (activity?.run { lifecycle.currentState.isAtLeast(Lifecycle.State.RESUMED) } == true) {
```

Comment:
I used the `run` to not have so many conditional checks. I don't know how smart the compiler is in that case. The latter option won't work, because `activity` is a getter that can't be captured in a closure.

Reply:
I find `?.run` inside of an `if` really ugly looking. Not really an issue though.

Reply:
I could also add another variable and cache the activity reference, then the null check would work (and do a smart cast).

Reply:
yes pretty code please!